NASA-AMMOS/aerie-ui
# Activity and Resource Browser

Refactors the Activity Types panel into the Activity and Resource Types panel and implements the following:
- Filter both activity and resource lists by multiple name filters
- Include `subsystem_tag` in activity type subscription
- Filter activity list by subsystem
- Filter resource list by data type
- Display data type and units (if found) in resource list
- Add all filtered items as a filter to a new or existing row (user prompted to select a layer within existing row)
- Add individual item as a filter to a new row or existing row
- Drag items into, above, and below existing rows to add as filters
- Create a new row using the "New Row +" button below the last row
- Added empty state when no rows exist in timeline
- Added a "Delete All" button to relevant sections of the Timeline Editor (rows, axes, guides, layers)
- Unified border and shadow styles for Menu, ContextMenu, and DatePicker
- Fix y axis label spacing and display bug for x-range layers
- Allow y axis ticks to be 0 instead of 1
- Update Chip styling
- Include subsystem tag in activity types subscription
- Improve tooltip exit animation duration
- Improved activity/resource style during drag

### Testing:
- Create some tags and manually associate various activity types to these tags in hasura
- Open the `Activity & Resource Types` panel and verify that the filtering and subsystem/data type dropdowns in activities/resources tabs filter the list below correctly
- Verify that items can be added to a layer or new row using the "Add Filter to Row" button
- Verify that an individual item can be added to a layer or new row using the (+) button found when hovering over an item
- Verify that all matching items or an individual item can be added to an existing row or inserted above/below a row
- Verify that all matching activities or an individual activity can be added to the plan by dragging them onto the canvas portion of the timeline
- Verify that activities are added into a single layer filter and that resources are added as individual layers with unique colors
- Verify that clicking the "New Row +" at the bottom of the timeline creates a new blank row
- Verify that rows, layers, axes, and guides can be bulk deleted from within the timeline editor and that the user is asked to confirm deletion

### TODO:
- [x] Review with team and users
- [x] Create individual layers when adding multiple resource layers
- [x] Affordance to indicate drag and drop action on items
- [x] Auto color for new layers if feasible
- [x] Prevent activity drag on timeline if user does not have permission to do so
- [x] Confirmation modal when adding an activity that will not be visible in the row
- [x] Confirmation modal before "delete all" for all timeline editor sections
- [x] Fix type issues and refactor in places

Closes #1416, closes #381

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/720e3b4c-8a0e-470e-96c3-50dc01426583">
